game/world/worldobjects.cpp

```diff
@@ -791,18 +791,18 @@ void WorldObjects::marchInteractives(DbgPainter &p) const {
     }
   }
 
-Interactive *WorldObjects::availableMob(const Npc &pl, std::string_view dest) {
-  const float  dist=100*10.f;
-  Interactive* ret =nullptr;
-
+Interactive *WorldObjects::findMob(const Npc &pl, std::string_view dest, float dist, const std::function<bool(const Interactive&)>& filter) {
   if(auto i = pl.interactive()){
     if(i->checkMobName(dest))
       return i;
     }
 
-  float curDist=dist*dist;
+  Interactive* ret     = nullptr;
+  float        curDist = dist*dist;
   interactiveObj.find(pl.position(),dist,[&](Interactive& i){
-    if(i.isAvailable() && i.checkMobName(dest)) {
+    if(!filter(i))
```

Comment:
Would be nice to call `filter` last, as caller side might assume that argument to filter is already matching scheme/distance